internal/tui/tui.go

```diff
@@ -30,16 +30,15 @@ func NewModel(b wifi.Backend) (*model, error) {
 	s.Spinner = spinner.Dot
 	s.Style = lipgloss.NewStyle().Foreground(CurrentTheme.Primary)
 
-	listModel := NewListModel()
-
 	m := model{
-		stack:         NewComponentStack(listModel),
 		spinner:       s,
 		backend:       b,
 		loading:       true,
 		statusMessage: "Loading connections...",
 	}
 	m.scanner = NewScanSchedule(func() tea.Msg { return scanMsg{} })
+	listModel := NewListModel(m.scanner)
```

Comment:
Instead of passing the m.scanner into NewListModel, seems we're mainly using it inside ListModel, should we initialize it and store it in there?